core/dbt/parser/unit_tests.py

```diff
@@ -187,10 +210,13 @@ def _get_original_input_node(self, input: str, tested_node: ModelNode):
                         name, package, version, self.manifest
                     )
             elif statically_parsed["sources"]:
-                input_package_name, input_source_name = statically_parsed["sources"][0]
-                original_input_node = self.manifest.source_lookup.find(
-                    input_source_name, input_package_name, self.manifest
-                )
+                for source in statically_parsed["sources"]:
```

Comment:
would we expect to get more than one result back from `statically_parsed["sources"]`? If so this loop could lead to surprising behaviour because original_input_node is overwritten on each iteration.

I'd expect that we should only ever get a single source back from this parsing because the input looks something like `source('test', 'test_name')`, in which case `source = statically_parsed["sources"][0]` would make more sense here conceptually

Reply:
That doesn't work because statically_parsed["sources"] is a set. So I did it the same way that the ref call above is handled. We could convert the set to a list and index it though, which might be better.

Reply:
ah, I see! yeah I think that's pretty subtle and converting to a list + grabbing the first (& only!) element would be more readable here

Reply:
Done.